build.rs

```diff
@@ -0,0 +1,12 @@
+extern crate protobuf_codegen_pure;
+
+fn main() {
+    if cfg!(feature = "gen") {
+        protobuf_codegen_pure::run(protobuf_codegen_pure::Args {
+            out_dir: "proto",
```

Comment:
Does not use the "proto", because it will modify the `metrics.rs` in the local registry, also if we build multiple projects that deps on the same version prometheus, the file may be corrupted due to multiple writers. Instead you can try `OUT_DIR`[0]. Here is an example [1].

[0] https://doc.rust-lang.org/cargo/reference/build-scripts.html#case-study-code-generation
[1] https://github.com/pingcap/tikv/pull/952

Reply:
The problem is the generated binding file contains a lot of module attributes, that means we can't use the `include!` macro to include the automatic generated binding file on the building time.  The `path` attribute could do it, but it doesn't accept a environment variable or `concat!` macro, it only accept the static relative path.

```rust
#[path = "../proto/metrics.rs"]
pub mod proto;
```

Reply:
Try `include!(concat!(env!("OUT_DIR"), "/metrics.rs"));` + `cfg_if! {}`. :wink:

Reply:
```rust
#[cfg(feature = "gen")]
pub mod proto {
    include!(concat!(env!("OUT_DIR"), "/metrics.rs"));
}
```
that's will cause a compile error.
> error: an inner attribute is not permitted following an outer attribute
  --> /Users/fl0120/github/rust-prometheus/target/debug/build/prometheus-52580fdca4c6c951/out/metrics.rs:12:3
   |
12 | #![allow(missing_docs)]
   |   ^
   |
   = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files. Outer attributes, like `#[test]`, annotate the item following them.